keychain/interface_test.go

```diff
@@ -304,6 +305,10 @@ func TestSecretKeyRingDerivation(t *testing.T) {
 		},
 	}
 
+	// We'll clamp the max range scan to constrain the run time of the
+	// test.
+	MaxKeyRangeScan = 3
```

Comment:
This seems like it could be racey, since it's modified in a parallel test. might be better to set this in an `init()` method in this file

Reply:
Fixed